parsons/salesforce/salesforce.py

```diff
@@ -0,0 +1,191 @@
+        logger.info(
+            f'Successfully updated {len(s)} out of {data_table.num_rows} records in {object}'
+        )
+        return r
+
+    def upsert(self, object, data_table, id_col):
+        """
+        Insert new records and update existing ones of the desired object in Salesforce
+
+        `Args:`
+            object: str
+                The API name of the type of records to upsert. Note that custom object names end
+                in `__c`
+            data_table: obj
+                A Parsons Table with data for upserting records. Column names must match object
+                field API names, though case and order need not match. Note that custom field
+                names end in `__c`.
+            id_col: str
+                The column name in `data_table` that stores the record ID. Required even if all
+                records are new/inserted.
+            `Returns:`
+                list of dicts that have the following data:
+                * success: boolean
+                * created: boolean (if new record is created)
+                * id: str (id of record created or altered, if successful)
+                * errors: list of dicts (with error details)
+        """
+
+        r = getattr(self.client.bulk, object).upsert(data_table.to_dicts(), id_col)
+        s = [x for x in r if x.get('success') is True]
+        logger.info(
+            f'Successfully upserted {len(s)} out of {data_table.num_rows} records to {object}'
+        )
+        return r
+
+    def delete(self, object, id_table, hard_delete=False):
+        """
+        Delete existing records of the desired object in Salesforce
+
+        `Args:`
+            object: str
+                The API name of the type of records to delete. Note that custom object names end
+                in `__c`
+            id_table: obj
+                A Parsons Table of record IDs to delete. Note that 'Id' is the default Salesforce
+                record ID field name.
+            hard_delete: boolean
+                If true, will permanently delete record instead of moving it to trash
+            `Returns:`
+                list of dicts that have the following data:
+                * success: boolean
+                * created: boolean (if new record is created)
+                * id: str (id of record deleted, if successful)
+                * errors: list of dicts (with error details)
+        """
+
+        if hard_delete:
+            r = getattr(self.client.bulk, object).hard_delete(id_table.to_dicts())
+        else:
+            r = getattr(self.client.bulk, object).delete(id_table.to_dicts())
+
+        s = [x for x in r if x.get('success') is True]
+        logger.info(
+            f'Successfully deleted {len(s)} out of {id_table.num_rows} records from {object}'
+        )
+        return r
+
+    @property
+    def client(self):
+        """
+        Get the Salesforce client to use for making all calls.
```

Comment:
You might want to include a link to the simple-sales-force documentation so folks can see other advanced methods that they can leverage.